spec/async/http/cache/general_spec.rb

```diff
@@ -68,43 +68,113 @@
 				else
 					Protocol::HTTP::Response[200, [['cache-control', 'max-age=1, public'], ['vary', 'user-agent']], ['Hello', ' ', 'World']]
 				end
-				
+
 				if request.head?
 					response.body = Protocol::HTTP::Body::Head.for(response.body)
 				end
-				
+
 				response
 			end
 		end
-		
+
 		let(:user_agents) {[
 			'test-a',
 			'test-b',
 		]}
-		
+
 		it "should cache GET requests" do
 			2.times do
 				user_agents.each do |user_agent|
 					response = subject.get("/", {'user-agent' => user_agent})
 					expect(response.headers['vary']).to include('user-agent')
-					expect(response.read).to be == user_agent
+					expect(response.read).to eq(user_agent)
 				end
 			end
-			
-			expect(store.index.size).to be 2
+
+			expect(store.index.size).to eq(2)
 		end
-	end
-	
+  end
+
+  context 'cache writes' do
+    context 'by response code' do
+      let(:app) do
+        Protocol::HTTP::Middleware.for do |_request|
+          Protocol::HTTP::Response[response_code, []] # no response body?
+        end
+      end
+
+      # [200, 203, 300, 301, 302, 404, 410].each do |response_code|
```

Comment:
The above list is accurate per the spec. Current code only caches 200, though. Should I leave it that way or change the code and uncomment the test to match the spec?

Reply:
I think it's acceptable to cache more response codes.

Reply:
Great, I'll extend it that way.

BTW, this could lead to a philosophical question about whether tests should be DRY with the implementation. I've definitely seen tests (including from new coworkers) that DRY up the tests+implementation by having a constant in the implementation and refer to that from the test. I don't believe in DRYing up code that much because it precludes TDD and the tests will always pass even if the constant gets changed. So instead I'll have a the list of cached response codes duplicated here and in the implementation.

Reply:
I’m fine with either approach.